go/src/koding/klient/machine/client/dynamic.go

```diff
@@ -146,7 +162,30 @@ func (dc *Dynamic) cron() {
 		pingTick    = time.NewTicker(dc.opts.PingInterval)
 	)
 
-	addr, err := machine.Addr{}, error(nil)
+	addr, ipAddr, err := machine.Addr{}, machine.Addr{}, error(nil)
+
+	// Lookup for IP address and set if changed.
+	setIP := func(a machine.Addr) {
+		// Do not lookup tunneled connections.
+		if _, err := dc.opts.AddrFunc("tunnel"); err == nil {
+			return
+		}
+
+		ipA, err := dc.opts.Builder.IP(a)
+		if err != nil || ipA.Network != "ip" {
```

Comment:
it should be possible to handle IPv6 too. And this logic does that(although I have never seen IPv6 in kite and I don't know if it handles it). IPv6 support also requires `-6` flag to `ssh` command and square brackets in rsync (`username@[IPv6]:~/dir`). We may want to create a separate issue for IPv6 support since it was never supported by kd.